python/lance/_lib.pyx

```diff
@@ -159,6 +159,10 @@ cdef class LanceFileFormat(FileFormat):
 def WriteTable(table: Table,
                sink: Union[str, Path],
                batch_size: int):
+    if not isinstance(table, Table):
```

Comment:
should this function accept the same types as 

https://arrow.apache.org/docs/python/generated/pyarrow.dataset.write_dataset.html ?

Reply:
It doesn't handle those other types does it?

Reply:
it as `write_dataset`? or `WriteTable`?

Reply:
this is our WriteTable. We can't customize write_dataset in pyarrow?